### DISCUSSION TITLE
Restrict auth when user has field 'disabled' set to true

### DISCUSSION THREAD

**Discussion by Arcitezz** [pocketbase/pocketbase#4350] [Q&A] [answered]
Hi, I'm not sure if this is possible but, I would like to restrict a user from authenticating with `.authWithPassword()`. The user that is trying to authenticate shouldn't be allowed to do so if their field `disabled` is set to `true` in the `users` auth collection.

I have tried to look for a way in the Admin UI, but couldn't find anything.

**Comment by ganigeorgiev** [comment, marked as answer]
There is no API rule for the authentication. You'll have to extend PocketBase with [Go](https://pocketbase.io/docs/go-event-hooks/) or [JS](https://pocketbase.io/docs/js-event-hooks/) hooks if you need custom auth behavior.

Here is an example using the JS [`onRecordAuthRequest`](https://pocketbase.io/docs/js-event-hooks/#onrecordauthrequest) hook:

```js
// pb_hooks/main.pb.js

onRecordAuthRequest((e) => {
    if (e.record.get("disabled")) {
        throw new ForbiddenError("You are not allowed to perform this action.")
    }
}, "users")
```

Alternatively, you can allow the "disabled" users to authenticate but add a check for the disabled state in the other collections (ex. `@request.auth.id != "" && @request.auth.disabled = false ...`).

**Reply by Arcitezz** [reply]
I must have overlooked the event hooks section in the docs, thanks for the quick reply and solution!
